pylib/openrazer/client/devices/mice.py

```diff
@@ -83,20 +84,88 @@ def dpi(self, value: tuple):
         if self.has('dpi'):
             if len(value) != 2:
                 raise ValueError("DPI tuple is not of length 2. Length: {0}".format(len(value)))
+            max_dpi = self.max_dpi
             dpi_x, dpi_y = value
 
             if not isinstance(dpi_x, int) or not isinstance(dpi_y, int):
                 raise ValueError("DPI X or Y is not an integer, X:{0} Y:{1}".format(type(dpi_x), type(dpi_y)))
 
-            if dpi_x < 0 or dpi_x > 16000:  # TODO add in max dpi option
+            if dpi_x < 0 or dpi_x > max_dpi:
                 raise ValueError("DPI X either too small or too large, X:{0}".format(dpi_x))
-            if dpi_y < 0 or dpi_y > 16000:  # TODO add in max dpi option
+            if dpi_y < 0 or dpi_y > max_dpi:
                 raise ValueError("DPI Y either too small or too large, Y:{0}".format(dpi_y))
 
             self._dbus_interfaces['dpi'].setDPI(dpi_x, dpi_y)
         else:
             raise NotImplementedError()
 
+    @property
+    def dpi_stages(self) -> list:
+        """
+        Get mouse DPI stages
+
+        Will return a list of tuples
+        :return: DPI stages [(500, 500), (1000, 1000), (2000, 2000) ...]
+        :rtype: list
+
+        :raises NotImplementedError: if function is not supported
+        """
+        if self.has('dpi_stages'):
+            raw_dpi_stages = self._dbus_interfaces['dpi'].getDPIStages()
+            dpi_stages = []
+
+            for dpi_x, dpi_y in raw_dpi_stages:
+                dpi_stages.append((int(dpi_x), int(dpi_y)))
+
+            return dpi_stages
+        else:
+            raise NotImplementedError()
+
+    @dpi_stages.setter
+    def dpi_stages(self, value: list):
+        """
+        Set mouse DPI stages
+
+        Daemon does type validation but can't be too careful
+        :param value: list of DPI X, Y tuple
+        :type value: list
+
+        :raises ValueError: If a tuple isn't long enough or contains invalid
+                            crap
```

Comment:
I know you copied this from above but it would be nice to reword it :D 
Maybe `:raises ValueError: when the input is invalid` ?

Reply:
Done in a21d86559e1603c32c1bfda8824dd98d75698f03